keras_cv/layers/feature_pyramid.py

```diff
@@ -191,6 +219,25 @@ def build_feature_pyramid(self, input_features):
         for level in reversed_levels:
             output_features[level] = self.output_layers[level](output_features[level])
 
+        # Build the extra level of FPN on top of existing input feature maps.
+        # This is used by cases like Retina-net
+        if self.num_extra_level:
+            for i in range(self.num_extra_level):
+                level = i + self.max_level + 1
+                # For the first extra level, it will use the feature map input from the
+                # top most pyramid level
+                if level == self.max_level + 1:
+                    inputs = input_features[self.max_level]
+                else:
+                    inputs = output_features[level - 1]
+
+                # Apply activations to all the levels except for the first one
```

Comment:
Hmmm...this seems quite specific, I wonder if it's possible to have a generic-fpn, and then a RetinaNetFPN (maybe even for a particular backbone) that deals with this?

Reply:
We could choose to fit all the logic for `num_extra_level` to a separate class, as long as we have a good name for it. 

Note that the generic FPN is having `num_extra_level` = 0, which totally skip this logic.

Reply:
We could take that approach, or have a RetinatNetFPN subclass the FPN. This will make the extra logic more separate from the base FPN.